meson.build

```diff
@@ -0,0 +1,49 @@
+project('hack-components', version: '0.0.0')
+
+conf = configuration_data()
+
+cmd = run_command('sh', '-c', 'echo $FLATPAK_ARCH')
+conf.set('FLATPAK_ARCH', cmd.stdout().strip())
+
+cmd = run_command('sh', '-c', 'echo $BRANCH')
+conf.set('BRANCH', cmd.stdout().strip())
+
+backgrounds_install_dir = join_paths(get_option('datadir'), 'backgrounds')
+conf.set('BACKGROUNDSDIR', join_paths(
+    '/var/lib/flatpak/app/com.endlessm.HackComponents',
```

Comment:
Something that occurred to me is that this won't work if HackComponents is installed in the user repository.

However we could easily make the control center support relative paths from the location of the XML file, which would make thins things a lot simpler here. You can probably just change this [1] to use `g_file_new_for_commandline_arg_and_cwd()` and pass the directory of the XML file as the cwd.

[1] https://github.com/endlessm/gnome-control-center/blob/17c90fdcb4ad62a1ef0964ee439c5b1f553a2388/panels/background/cc-background-xml.c#L209